osquery/tables/networking/linux/tests/namespace_tests.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <gtest/gtest.h>
+
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+
+#include "osquery/tests/test_additional_util.h"
+#include "osquery/tests/test_util.h"
+
+#include <netinet/in.h>
+#include <sched.h>
+#include <signal.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+namespace {
+const int kNamespaceTestPort = 27960;
+
+// It is easier to allocate these structures here instead of inside the
+// new entry point (which may cause stack issues)
+struct sockaddr_in server_addr = {};
```

Comment:
What kinds of stack issues were you seeing? Any time we have global level variables like this it makes me a bit nervous as they're not `const`.

Reply:
I am allocating the stack manually due to how the clone() syscall works with the flags I'm passing; I noticed that if I start doing too many things it will crash